### PR TITLE
ROF data moab driver

### PR DESCRIPTION
**Pull request by iulian787**
Extend moab driver for a case with ROF data 

 Case tested: --compset GMPAS-IAF --res T62_oQU240wLI

iMOAB_MergeVertices is not used anymore in moab land driver, small correction incorporated into this PR

-------

Update on Jun 12, 2024: I rebased the branch [iulian787/moab_data_rof](https://github.com/E3SM-Project/E3SM/tree/iulian787/moab_data_rof)  off Jon Wolfe's branch, that introduces corrected files:
[jonbob/cpl/fix-t62-files] (https://github.com/E3SM-Project/E3SM/tree/jonbob/cpl/fix-t62-files)

https://github.com/E3SM-Project/E3SM/pull/6459

This now works with moab master, tested on gce machine at Argonne

### CODE REVIEW

**Comment by rljacob** [comment]
@jonbob should we switch to the new domain file as the default?  Do you know the history of why it doesn't have a pole at 90N?

File: components/elm/src/cpl/lnd_comp_mct.F90 [hunk 1]
```diff
@@ -846,7 +846,7 @@ subroutine init_moab_land(bounds, LNDID)
      use spmdmod          , only: masterproc
     use iMOAB        , only: iMOAB_CreateVertices, iMOAB_WriteMesh, iMOAB_RegisterApplication, &
     iMOAB_DefineTagStorage, iMOAB_SetIntTagStorage, iMOAB_SetDoubleTagStorage, &
-    iMOAB_ResolveSharedEntities, iMOAB_CreateElements, iMOAB_MergeVertices, iMOAB_UpdateMeshInfo
```
Comment (rljacob) [inline components/elm/src/cpl/lnd_comp_mct.F90:849]: Also mention these changes in the PR description.

**Comment by jonbob** [comment]
@rljacob -- let me look

**Comment by rljacob** [comment]
Jon has mapping weights files ready.

**Comment by rljacob** [comment]
Will merge after #6459

**Comment by rljacob** [comment]
Please update the PR description with the text you want in the merge commit.

**Review by rljacob — APPROVED** [review]

**Pull request merged**